### PR TITLE
finished all logic functions, fix all errors with TS and ESlint, add …

### PR DESCRIPTION
**Pull request by Egorka0105**
…all style

### CODE REVIEW

File: my-app/src/Components/Table/TableAvg/TableAvg.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { nanoid } from 'nanoid';
+import styles from './tableAvg.module.scss';
+import { RootState } from '../../../store/store';
+import { findAvg } from '../../../services/findAvg';
+
+function TableAvg() {
+	const avg = useSelector((state: RootState) => state.tableStore.table);
+	const avgArr = findAvg(avg);
```
Comment (ZamkovaTetiana) [inline my-app/src/Components/Table/TableAvg/TableAvg.tsx]: use `useMemo` for better performance

File: my-app/src/Components/Table/TableAvg/TableAvg.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+						className={styles.td}
+						style={
+							el.percentActive
+								? {
+										background: `linear-gradient(0deg, red ${el.percent}%, white ${el.percent}%)`,
+										color: '#000',
+								  }
+								: el.nearest
+								? { background: '#f8ff3e' }
+								: { background: '#56FC7CFF' }
```
Comment (ZamkovaTetiana) [inline my-app/src/Components/Table/TableAvg/TableAvg.tsx]: rework with CSS classes

File: my-app/src/services/cells.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { obj } from './interfaces';
+
+export const cells = (arr, currentObj) => {
```
Comment (ZamkovaTetiana) [inline my-app/src/services/cells.tsx:3]: //

**Review by ZamkovaTetiana — COMMENTED** [review]
Need to review it all together with you

File: my-app/src/Components/Table/TableAvg/TableAvg.tsx (hunk 1 above)
Comment (Egorka0105) [reply my-app/src/Components/Table/TableAvg/TableAvg.tsx]: edited

File: my-app/src/Components/Table/TableAvg/TableAvg.tsx (hunk 2 above)
Comment (Egorka0105) [reply my-app/src/Components/Table/TableAvg/TableAvg.tsx]: edited

File: my-app/src/services/cells.tsx (hunk 3 above)
Comment (Egorka0105) [reply my-app/src/services/cells.tsx:3]: edited

**Pull request merged**